### PR TITLE
Added Amine's in tandem intersection code.

### PR DESCRIPTION
**Pull request by shahidkhaliq**

### CODE REVIEW

**Comment by queryproc** [comment]
Sign the CLA.

**Comment by shahidkhaliq** [comment]
Signed it @queryproc

File: src/main/java/ca/waterloo/dsg/graphflow/graph/SortedAdjacencyList.java [hunk 1]
```diff
@@ -215,34 +215,48 @@ public void removeNeighbour(int neighbourId, short edgeTypeFilter) {
...
+                }
+            } else if (neighbourIds[thisIndex] > otherAdjList.get(otherIndex)) {
+                otherIndex++;
+                while (otherIndex < otherAdjList.getSize() &&
+                        neighbourIds[thisIndex] > otherAdjList.get(otherIndex)) {
+                    otherIndex++;
+                }
             } else {
-                index = (resultIndex) > -1 ? resultIndex : -resultIndex;
+                if (edgeTypeFilter == -1 || edgeTypeFilter == edgeTypes[thisIndex]) {
```
Comment (queryproc) [inline src/main/java/ca/waterloo/dsg/graphflow/graph/SortedAdjacencyList.java]: Change this -1 with TypeAndPropertyFilter.ANY

File: src/main/java/ca/waterloo/dsg/graphflow/graph/SortedAdjacencyList.java [hunk 2]
```diff
@@ -215,34 +215,48 @@ public void removeNeighbour(int neighbourId, short edgeTypeFilter) {
...
+                }
+            } else if (neighbourIds[thisIndex] > otherAdjList.get(otherIndex)) {
+                otherIndex++;
+                while (otherIndex < otherAdjList.getSize() &&
+                        neighbourIds[thisIndex] > otherAdjList.get(otherIndex)) {
+                    otherIndex++;
+                }
             } else {
-                index = (resultIndex) > -1 ? resultIndex : -resultIndex;
+                    if (edgeTypeFilter == TypeAndPropertyKeyStore.ANY || edgeTypeFilter == edgeTypes[thisIndex]) {
```
Comment (queryproc) [inline src/main/java/ca/waterloo/dsg/graphflow/graph/SortedAdjacencyList.java]: This line is more than 100 chars which is the limit per line that we follow, return to line after the "||".

File: src/main/java/ca/waterloo/dsg/graphflow/graph/SortedAdjacencyList.java [hunk 3]
```diff
@@ -215,34 +215,48 @@ public void removeNeighbour(int neighbourId, short edgeTypeFilter) {
...
+        } else {
+            intersection = new IntArrayList(otherAdjList.getSize());
+        }
+
+        int thisIndex = 0;
+        int otherIndex = 0;
+        while ((thisIndex < size) && (otherIndex < otherAdjList.getSize())) {
+            if (neighbourIds[thisIndex] < otherAdjList.get(otherIndex)) {
+                thisIndex++;
+                while (thisIndex < size && neighbourIds[thisIndex] < otherAdjList.get(otherIndex)) {
```
Comment (queryproc) [inline src/main/java/ca/waterloo/dsg/graphflow/graph/SortedAdjacencyList.java]: Double check that this while is exactly 100 chars as it seems like it but be sure.

File: src/main/java/ca/waterloo/dsg/graphflow/util/IntArrayList.java [hunk 4]
```diff
@@ -37,6 +37,15 @@ public void add(int element) {
         data[size++] = element;
     }
 
+    /**
+     * Adds a new integer to the list without resizing the list.
```
Comment (queryproc) [inline src/main/java/ca/waterloo/dsg/graphflow/util/IntArrayList.java]: add 'without resizing the list assuming the capacity needed has been preallocated.

**Pull request merged**